### PR TITLE
Update module file after Gaea-C5 OS upgrade

### LINKED ISSUE #1071: UPP fails to build on Gaea-C5 after OS upgrade
The EPIC SI team has reinstalled spack-stack-1.6.0 at the same location it existed before. 2 spack-stack module versions will need to be increased to build upp:
stack-intel 2023.1.0 to 2023.2.0
stack-cray-mpich 8.1.25 to 8.1.28

### PR DESCRIPTION
**Pull request by DavidBurrows-NCO**
The EPIC SI team has reinstalled spack-stack-1.6.0 at the same location it existed before. This PR updates version numbers in modulefiles/gaea.lua:
stack-intel 2023.1.0 to 2023.2.0
stack-cray-mpich 8.1.25 to 8.1.28

Resolves #1071 
Refs NOAA-EMC/global-workflow [#3011](https://github.com/NOAA-EMC/global-workflow/issues/3011)

### CODE REVIEW

**Comment by WenMeng-NOAA** [comment]
@DavidBurrows-NCO Please open a new [issue](https://github.com/NOAA-EMC/UPP/issues) linking to this PR. Are you working at EMC, EPIC or another affiliate?

**Comment by DavidBurrows-NCO** [comment]
@WenMeng-NOAA I added the issue and global-workflow reference to the original PR comment.  My main affiliation is EPIC, but this is work done alongside @aerorahul's global-workflow team.

**Comment by DavidBurrows-NCO** [comment]
@WenMeng-NOAA @aerorahul Since the ufs-wx-model ([Ref #2448](https://github.com/ufs-community/ufs-weather-model/pull/2448#discussion_r1806540819)) is setting up the gaeac5 and gaeac6 convention, should I go ahead and set that up for upp and other global-workflow submodules, i.e., convert all current gaea references to gaeac5.  Follow up PRs can handle the C6 build/runs.

**Comment by WenMeng-NOAA** [comment]
> @WenMeng-NOAA @aerorahul Since the ufs-wx-model ([Ref #2448](https://github.com/ufs-community/ufs-weather-model/pull/2448#discussion_r1806540819)) is setting up the gaeac5 and gaeac6 convention, should I go ahead and set that up for upp and other global-workflow submodules, i.e., convert all current gaea references to gaeac5. Follow up PRs can handle the C6 build/runs.

@DavidBurrows-NCO I would suggest adding modulefile updates for both Gaea C5 and C6 in UPP, so they will be propagate to UFSWM with the upp submodule upgrade.

**Comment by DavidBurrows-NCO** [comment]
@WenMeng-NOAA I have updated this PR to include building on C6 and differentiated detect machine and modulefiles between gaeac5 and gaeac6.  I re-cloned and tested builds on both systems. Let me know your thoughts. Thanks.

File: modulefiles/gaeac6.lua [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+]])
+
+whatis([===[Loads libraries needed for building the UPP on Gaea C6 ]===])
+
+prepend_path("MODULEPATH", "/autofs/ncrc-svm1_proj/epic/spack-stack/spack-stack-1.6.0/envs/upp-addon-env/install/modulefiles/Core")
+
+stack_intel_ver=os.getenv("stack_intel_ver") or "2023.2.0"
+load(pathJoin("stack-intel", stack_intel_ver))
+
+stack_cray_mpich_ver=os.getenv("stack_cray_mpich_ver") or "8.1.28"
```
Comment (WenMeng-NOAA) [inline modulefiles/gaeac6.lua:13]: @DavidBurrows-NCO The UFSWM PR #[2448](https://github.com/ufs-community/ufs-weather-model/pull/2448) uses 8.1.29. Would you like to keep the version consistent with UFSWM?

**Comment by WenMeng-NOAA** [comment]
@DavidBurrows-NCO I don't have Gaea access. Currently, the UPP RTs don't include Gaea. As long as you have completed the UPP executable build testing, it is fine with me.

File: modulefiles/gaeac6.lua (hunk 1 above)
Comment (DavidBurrows-NCO) [reply modulefiles/gaeac6.lua:13]: @WenMeng-NOAA I checked into using `stack-cray-mpich.v8.1.29` on Gaea-C6.  To do that, I would have to switch `spack-stack` environment to `unified-env-c6`, however, `unified-env-c6` only contains `g2.v3.4.5` and `g2tmpl.v1.10.2`.

Comment (WenMeng-NOAA) [reply modulefiles/gaeac6.lua:13]: @DavidBurrows-NCO Thanks for checking and explanation. In that case, we keep the current settings for building offline post.

**Comment by WenMeng-NOAA** [comment]
The UPP RTs have been completed on WCOSS2 with no baseline changes.

**Review by WenMeng-NOAA — APPROVED** [review]

**Comment by WenMeng-NOAA** [comment]
This PR is ready for merging.

**Comment by DavidBurrows-NCO** [comment]
Thanks for your help @WenMeng-NOAA

**Pull request merged**